css/index.css

```diff
@@ -86,25 +102,29 @@ p {
 }
 
 .middle-content {
+    display: flex;
+    flex-wrap: wrap;
     margin-bottom: 20px;
     border-bottom: 1px dashed black;
 }
 
 .middle-content h2 {
+    display: flex;
     padding: 0 2%;
     margin-bottom: 0;
 }
 
```

Comment:
One thing that may help here to get the boxes sized properly is specifying the size of the div.